hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala

```diff
@@ -509,12 +510,22 @@ abstract class HoodieBaseRelation(val sqlContext: SQLContext,
     StructType(dataStructSchema.filterNot(f => partitionColumns.contains(f.name)))
 
   private def isSchemaEvolutionEnabled = {
+    // Auto set schema evolution
+    // Wrapped as a function, this function is triggered only when called, minimize the time cosumption.
+    def detectSchemaEvolution(): Boolean = {
+      val result = new FileBasedInternalSchemaStorageManager(metaClient).isValidHistorySchemaExist
```

Comment:
is it possible to avoid creating a new FileBasedInternalSchemaStorageManager every check? the actual info is retrieved by the existing metaclient, so `isValidHistorySchemaExist` can be a static util taking the metaclient?. This would involve more refactoring work so it's optional here.